core/src/std_packages.rs

```diff
@@ -176,44 +289,172 @@ pub fn native_block_content<T, U>(
+                Ok(Element::Compound(vec![]))
+            } else {
+                Ok(text_element(default.to_string(), id.clone()))
+            }
+        }
+    }
+}
+
+/// Push a value to a list
+pub fn list_push<T, U>(
+    ctx: &mut Context<T, U>,
+    value: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    _: &OutputFormat,
+    _: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+    ctx.state.variables.list_push(name, value)?;
+    Ok(Element::Compound(vec![]))
+}
+
+/// Read a list of values
+pub fn list_read<T, U>(
+    ctx: &mut Context<T, U>,
+    _: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    format: &OutputFormat,
+    id: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+
+    match ctx.state.variables.get(name) {
+        Some(value @ variables::Value::List(_)) => Ok(text_element(value.to_string(), id.clone())),
+        Some(value) => Err(CoreError::TypeMismatch {
+            name: name.to_string(),
+            expected_type: VarType::List,
+            present_type: value.get_type(),
+        }),
+        None => {
+            let default = args.get("default").map(ArgValue::as_str).flatten().unwrap();
+            if default == "<log error>" {
+                // No default was provided as fallback, let's log a error
+                log_var_error(ctx, name, "list-read", format);
+                Ok(Element::Compound(vec![]))
+            } else {
+                Ok(text_element(default.to_string(), id.clone()))
+            }
+        }
     }
 }
 
+/// Add a string to a set
+pub fn set_add<T, U>(
+    ctx: &mut Context<T, U>,
+    value: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    _: &OutputFormat,
+    _: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+    ctx.state.variables.set_add(name, value)?;
+    Ok(Element::Compound(vec![]))
+}
+
+/// Read a list of values
+pub fn set_read<T, U>(
+    ctx: &mut Context<T, U>,
+    _: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    format: &OutputFormat,
+    id: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+
+    match ctx.state.variables.get(name) {
+        Some(value @ variables::Value::Set(_)) => Ok(text_element(value.to_string(), id.clone())),
+        Some(value) => Err(CoreError::TypeMismatch {
+            name: name.to_string(),
+            expected_type: VarType::Set,
+            present_type: value.get_type(),
+        }),
+        None => {
+            let default = args.get("default").map(ArgValue::as_str).flatten().unwrap();
+            if default == "<log error>" {
+                // No default was provided as fallback, let's log a error
+                log_var_error(ctx, name, "set-read", format);
```

Comment:
This should generate an `[error]` node to facilitate inline error reporting and keep our convention